cmd/dep/testdata/harness_tests/init/case4/final/Gopkg.toml

```diff
@@ -0,0 +1,8 @@
+
+[[constraint]]
+  name = "github.com/sdboyer/deptest"
+  version = "1.0.0"
```

Comment:
Previously, this used to be with a `^`. Now, I think because of `ImpliedCaretString` versions are written as exact. Is it intended?
Also, the feedback that we print for network mode, prints with `^` right now. So there's a mismatch between what the feedback shows version and what's written in manifest.